src/screens/registerStudent/hooks/types.ts

```diff
@@ -0,0 +1,97 @@
+
+  password: string;
+  passwordError: string;
+  showPassword: boolean;
+  handlePasswordChange(e: React.ChangeEvent<HTMLInputElement>): void;
+
+  confirmPassword: string;
+  confirmPasswordError: string;
+  handleConfirmPasswordChange(e: React.ChangeEvent<HTMLInputElement>): void;
+
+  handleClickShowPassword(): void;
+  handleMouseDownPassword(e: React.MouseEvent<HTMLButtonElement>): void;
+  handleContinueClick(e: React.SyntheticEvent<EventTarget>): void;
+  handleCancelRegisterClick(): void;
+};
+
+export type TMiddleStudentRegisterHook = {
+  courses: TCourseData[];
+  enrollment: string;
+  enrollmentError: string;
+  handleEnrollmentChange(e: React.ChangeEvent<HTMLInputElement>): void;
+
+  course: string;
+  handleCourseChange(e: SelectChangeEvent<unknown>): void;
+
+  description: string;
+  handleDescriptionChange(e: React.ChangeEvent<HTMLInputElement>): void;
+
+  handleBackClick(): void;
+  handleSecondContinueClick(e: React.SyntheticEvent<EventTarget>): void;
+};
+
+export type TFinalStudentRegisterHook = {
+  contactEmail: string;
+  handleContactEmailChange(e: React.ChangeEvent<HTMLInputElement>): void;
+
+  linkedin: string;
+  handleLinkedinChange(e: React.ChangeEvent<HTMLInputElement>): void;
+
+  whatsapp: string;
+  handleWhatsappChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleBackClick(): void;
+
+  isLoading: boolean;
+  handleFinalClick(): void;
+};
+
+export type TStudentRegisterHook = {
+  step: number;
+  courses: TCourseData[];
+  course: string;
+  name: string;
+  enrollment: string;
+  description: string;
+  email: string;
+  contactEmail: string;
+  linkedin: string;
+  whatsapp: string;
+  password: string;
+  confirmPassword: string;
+  showPassword: boolean;
+  nameError: string;
+  enrollmentError: string;
+  emailError: string;
+  passwordError: string;
+  confirmPasswordError: string;
+  isLoading: boolean;
+  courseId: number;
+  handleContinueClick(e: React.SyntheticEvent<EventTarget>): void;
+  handleBackClick(): void;
+  handleSecondContinueClick(e: React.SyntheticEvent<EventTarget>): void;
+  handleClickShowPassword(): void;
+  handleMouseDownPassword(e: React.MouseEvent<HTMLButtonElement>): void;
+  handleContactEmailChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleLinkedinChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleWhatsappChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleCourseChange(e: SelectChangeEvent<unknown>): void;
+  handleDescriptionChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleEnrollmentChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleNameChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleEmailChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handlePasswordChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleConfirmPasswordChange(e: React.ChangeEvent<HTMLInputElement>): void;
+  handleFinalClick(): void;
+  handleCancelRegisterClick(): void;
+};
```

Comment:
Como esse tipo é a resultado da junção dos três tipos acima, você pode usar interceção de types para defini-lo. Segue **[aqui](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types)** a referência.
```suggestion
export type TStudentRegisterHook = TFinalStudentRegisterHook & TMiddleStudentRegisterHook & TStartStudentRegisterHook;
```

Reply:
Chocado demais amiga, abalou todas as estruturas. Aderido